src/multi/bmp180.cpp

```diff
@@ -1,16 +1,20 @@
 #include <libhal-sensor/multi/bmp180.hpp>
 
+#include "bmp180_internal.hpp"
+
 #include <array>
 
+#include <libhal-util/bit.hpp>
+#include <libhal-util/enum.hpp>
 #include <libhal-util/i2c.hpp>
 #include <libhal/error.hpp>
 
 using namespace std::literals;
 namespace hal::sensor {
 namespace {
 
-// functions for creating arrays containing the different register addresses in
-// their payloads
+// Helper functions for creating arrays containing the different register
+// addresses in their payloads.
 auto device_id_register()
```

Comment:
```suggestion
// Helper functions for creating arrays containing the different register
// addresses in their payloads.
constexpr auto device_id_register()
```

Make all of the helper functions that just return constant information constexpr to give the compiler more of an opportunity to optimize these.